lua/controller.lua

```diff
@@ -0,0 +1,377 @@
+--[[
+Minetest Mod Storage Drawers - A Mod adding storage drawers
+
+Copyright (C) 2018 isaiah658
+Copyright (C) 2018 LNJ <git@lnj.li>
+
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+]]--
+
+-- The gist of how the drawers mod stores data is that there are entities and the drawer node itself. The entities are needed to allow having multiple drawers in one node. The entities and node each store metadata about the item counts and such. It is necessary to change both at once otherwise in some cases the entity values are used and in other cases the node metadata is used.
+
+-- The gist of how the controller works is this. The drawer controller scans the adjacent tiles (length and height is configurable) and puts the item names and other info such as coordinates and the visualid of the entity in a table. That table is saved in the controllers metadata. The table is used to help prevent needing to scan all the drawers to deposit an item in certain situations. The table is only updated on an as needed basis, not by a specific time/interval. Controllers that have no items will not continue scanning drawers.

```

Comment:
Please only use about 80 chars per line. And for this I'd suggest you use a `--[[ ]]--` construct.